app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt

```diff
@@ -133,72 +129,22 @@ private fun Username(username: String) {
 }
 
 @Composable
-private fun MessageContent(messageContent: MessageContent) {
+private fun MessageContent(messageContent: MessageContent, onAssetClick: (String) -> Unit) {
     when (messageContent) {
         is MessageContent.ImageMessage -> MessageImage(
             rawImgData = messageContent.rawImgData,
             imgParams = ImageMessageParams(messageContent.width, messageContent.height)
         )
         is MessageContent.TextMessage -> MessageBody(messageBody = messageContent.messageBody)
-    }
-}
-
-@Composable
-fun MessageImage(rawImgData: ByteArray?, imgParams: ImageMessageParams) {
-    Image(
-        painter = rememberAsyncImagePainter(
-            rawImgData?.toBitmap() ?: getUriFromDrawable(
-                LocalContext.current,
-                R.drawable.ic_gallery
-            )
-        ),
-        alignment = Alignment.CenterStart,
-        contentDescription = stringResource(R.string.content_description_image_message),
-        modifier = Modifier.width(imgParams.normalizedWidth).height(imgParams.normalizedHeight),
-        contentScale = ContentScale.Crop
-    )
-}
-
-class ImageMessageParams(private val realImgWidth: Int, private val realImgHeight: Int) {
-    // Image size normalizations to keep the ratio of the inline message image
-    val normalizedWidth: Dp
-        @Composable
-        get() = MaterialTheme.wireDimensions.messageImageMaxWidth
-
-    val normalizedHeight: Dp
-        @Composable
-        get() = Dp(normalizedWidth.value * realImgHeight.toFloat() / realImgWidth)
-}
-
-// TODO: Here we actually need to implement some logic that will distinguish MentionLabel with Body of the message,
-// waiting for the backend to implement mapping logic for the MessageBody
-@Composable
-private fun MessageBody(messageBody: MessageBody) {
-    Text(
-        buildAnnotatedString {
-            appendBody(messageBody = messageBody)
-        }
-    )
-}
-
-// TODO:we should provide the SpanStyle by LocalProvider to our Theme, later on
-@Composable
-private fun AnnotatedString.Builder.appendMentionLabel(label: String) {
-    withStyle(
-        style = SpanStyle(
-            color = MaterialTheme.colorScheme.primary,
-            background = MaterialTheme.colorScheme.primaryContainer,
+        is MessageContent.AssetMessage -> MessageAsset(
+            assetName = messageContent.assetName.split(".").first(),
```

Comment:
I wonder what would happen when the name of the asset has more than one dot 🤔  I think it will get an uncompleted name. Maybe we could use something like the following to be safe:

```suggestion
            assetName = messageContent.split(".")
            .dropLast(1)
            .joinToString("."),
```

Reply:
Yepp, that sounds better actually. However there are also certain extensions that include also several dots, like for example "some-mock-asset.tar.gz". In this case this parsing will not work either. And the only way I can think of to fix those ones would be to keep some sort of file extensions list to compare against.
But for the moment I think we can live with your suggested change, and revisit it later on if anyone ever complains or if we detect this causes any major issues.